pkgs/development/libraries/agda/agda-categories/default.nix

```
@@ -12,10 +12,10 @@ mkDerivation rec {
   };
 
   # Remove this once new version of agda-categories is released which
-  # directly references standard-library-1.7.1
+  # directly references standard-library-1.7.2
   postPatch = ''
     substituteInPlace agda-categories.agda-lib \
-      --replace 'standard-library-1.7' 'standard-library-1.7.1'
+      --replace 'standard-library-1.7' 'standard-library-1.7.2'
```

Agda's [version matching system](https://agda.readthedocs.io/en/latest/tools/package-system.html#version-numbers) is somewhat insane: there's no middle ground between "no version constraint" and "exact match down to patch". In order to avoid having to change this with every standard-library update I think we should drop the constraint altogether:

```suggestion
      --replace 'standard-library-1.7' 'standard-library'
```

But beware that the left hand side changes in later versions of agda-categories, so we really want something like

```sh
sed -Ei 's/standard-library-[0-9.]+/standard-library/' agda-categories.agda-lib
```

Maybe even convince upstream to drop the constraint.

Updating agda-categories will take some work anyway, from what I can tell (Everything.agda has to be generated, which seems to involve git).

---

Good idea! In the imminent rebase and force-push, I include your sed suggestion.

I am not entirely sure what you mean by the comment on Everything.agda. Currently we don't do anything ourselves regarding Everything.agda and everything seems fine :-)

---

Right, but future updates will not be fine. See https://github.com/agda/agda-categories/commit/8643687021b321420665bdf7f9a27983b564291b

---

Ah, I see! Thank you for pointing this out.

In the next force-push, I will upgrade to categories v0.1.7.1a (thereby exhibiting the issue you raised) and fix it by generating Everything.agda ourselves.

The command listed in the Makefile doesn't seem suitable for us, because we are not working in a Git checkout. But the following command, added to postPatch, seems to do its job just fine:

```
find src -name '*.agda' | sed -e 's|^src/[/]*|import |' -e 's|/|.|g' -e 's/.agda//' -e '/import Everything/d' | LC_COLLATE='C' sort > Everything.agda
```

Compilation is still running, I will report once it finishes.